src/helpers.ts

```diff
@@ -3,14 +3,25 @@ import request from "sync-request";
 import * as bip32 from 'bip32';
 import chalk from "chalk";
 
-import { BITCOIN_NETWORK, LITECOIN_NETWORK, network } from "./settings";
-import { transientLine } from "./display";
+import { 
+  BITCOIN_NETWORK, 
+  LITECOIN_NETWORK,
+  configuration 
+} from "./settings";
 
 // TODO: properly rework this function
-function getJSON(url: string, attempts = 0) {
-  const res = request('GET', url);
+function getJSON(url: string, APIKey?: string) {
+  let headers = {};
 
-  if (attempts > 5) {
+  if (APIKey !== undefined) {
+    headers = {
+      'X-API-Key': APIKey
```

Comment:
While this is meant to be generic, there is no guarantee other provides will use this header name.

Reply:
Indeed. This whole function will have to be refactored in the future (including making it async, handle retries, etc.).

Reply:
I suggest that we keep that way at the moment, and refactor it completely later. What do you think @YBadiss?